### PR TITLE
refactor: move option parsing into common.wrap()

### PR DESCRIPTION
**Pull request by nfischer**
This is roughly in-line with what was proposed for the plugin API. This moves option parsing as part of `common.wrap()`, so contributors don't need to worry about making the call themselves. By default, `common.wrap()` acts as it did before, and will only parse options if options are specified when the plugin is registered.

For trickier commands, like `chmod()` and `set()`, which sometimes need special parsing, the contributor can choose to parse these commands with an explicit call to `common.parseOptions()`.

This deviates from the plugin API draft in that command options are passed in via the `cmdOptions` attribute, which I believe is more descriptive than an anonymous parameter to `common.register()`. This also makes it easier to leave option parsing optional.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -318,6 +318,9 @@ introduced between each file).
 
 ### head([{'-n', \<num\>},] file [, file ...])
 ### head([{'-n', \<num\>},] file_array)
+Available options:
```
Comment (ariporad) [inline README.md]: Why these changes? They seem unrelated.

Comment (nfischer) [inline README.md]: @ariporad As I was digging through, I saw that the documentation had an error (it was missing the available option). You're correct, these changes are unrelated to the refactor.

Comment (ariporad) [inline README.md]: kk. LGTM

**Comment by ariporad** [comment]
LGTM
